get_s3.go

```diff
@@ -313,7 +313,7 @@ func (g *S3Getter) parseUrl(u *url.URL) (region, bucket, path, version string, c
 	} else {
 		pathParts := strings.SplitN(u.Path, "/", 3)
 		if len(pathParts) != 3 {
-			err = fmt.Errorf("URL is not a valid S3 compliant URL")
+			err = fmt.Errorf("URL is not a valid S3 URL")
```

Comment:
I **_think_** we have precedence for breaking/changing error message in the past. But, because we don't provide nicer error value structs/types, consumers of this library would need to do string matching. 

Do we *need* to change this?

Reply:
That's a good catch, I forgot about this change it was to ease the unit test for `Test_S3Getter_ParseUrl_Malformed`. Should be like before, adjusted the tests in https://github.com/hashicorp/go-getter/pull/512/commits/4836732617d466dac708e19e5023293afa2e49b6